### PR TITLE
Make sub nav bar aware of react-router-dom

### PR DESCRIPTION
**Pull request by mfrachet**
- Make sub nav bar aware of react-router-dom
- Add a border to the sub nav bar
- Add a pages in the CRA example

An example of a settings page

### CODE REVIEW

**Comment by HichamELBSI** [comment]
@mfrachet I just saw something on the main nav

Screenshot 2021-07-20 at 15 23 41

**Comment by mfrachet** [comment]
haha, you're right! I'll update the main nav on another PR, thanks for the catch 🤣

**Comment by HichamELBSI** [comment]
I think we need a `z-index` on the condensed menu icon and the avatar is not centered vertically.

Screenshot 2021-07-20 at 15 30 34

**Comment by mfrachet** [comment]
I've made the fixes @HichamELBSI 

Screenshot 2021-07-20 at 15 39 13
Screenshot 2021-07-20 at 15 39 16

File: example/src/pages/SettingsPage.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                Edit
+              </Button>
+            }
+            title="Some settings"
+            subtitle="36 entries found"
+            as="h2"
+          />
+        }
+      >
+        <Box padding={6}>Hello</Box>
```
Comment (soupette) [inline example/src/pages/SettingsPage.js:33]: Can you add a Switch here?

Something like:

```
<Switch>
<Route path="/settings/application">
  <Box padding={6}>Hello</Box>
</Route>
<Route path="">
  <Box paddin={6}> TODO</Box>
</Route>
</Switch>
```

File: example/src/shared/SettingsNav/index.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+} from "@strapi/design-system";
+
+export const SettingsNav = () => {
+  const [search, setSearch] = useState("");
+  const links = [
+    {
+      id: 1,
+      label: "Addresses",
+      icon: <AlertWarningIcon />,
+      to: "/address",
```
Comment (soupette) [inline example/src/shared/SettingsNav/index.js:18]: Can you prefix all the `to` with `/settings/` ?

File: example/src/shared/SettingsNav/index.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        onChange={(e) => setSearch(e.target.value)}
+        label="Settings"
+        searchLabel="Search..."
+      />
+      <SubNavSections>
+        <SubNavLink
+          to="/blabla"
+          withBullet
+          icon={<Applications />}
+          className="active"
```
Comment (soupette) [inline example/src/shared/SettingsNav/index.js:54]: I think we can remove this prop

File: example/src/shared/SettingsNav/index.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        searchable
+        value={search}
+        onClear={() => setSearch("")}
+        onChange={(e) => setSearch(e.target.value)}
+        label="Settings"
+        searchLabel="Search..."
+      />
+      <SubNavSections>
+        <SubNavLink
+          to="/blabla"
```
Comment (soupette) [inline example/src/shared/SettingsNav/index.js:51]: Here with the previous comment it should be `/settings/application`

File: example/src/App.js [hunk 5]
```diff
@@ -9,10 +10,15 @@ function App() {
         <Route path="/cm">
           <CM />
         </Route>
+
+        <Route path="/cm/edit">
+          <EditViewPage />
+        </Route>
+
+        <Route path="/settings">
```
Comment (soupette) [inline example/src/App.js:18]: I think the path should be `/settings/:settingType` so you can redirect the user to the application setting when the user clicks on the Setting link from the main menu by using en effect in the SettingsPage or another component.

**Review by soupette — CHANGES_REQUESTED** [review]

**Review by soupette — APPROVED** [review]
LGTM!

**Pull request merged**